### PR TITLE
Bump rustc to 1.65

### PR DESCRIPTION
**Pull request by PicoCentauri**
----
:books: Documentation preview :books:: https://rascaline--245.org.readthedocs.build/en/245/

### CODE REVIEW

**Comment by Luthaf** [comment]
This is a bit weird. CI in #235 is failing due to regex, but we should not have regex anywhere in our dependency tree. I'll look a bit more into this!

**Comment by PicoCentauri** [comment]
`cargo tree` told me it seems to live in a dev dependency of `criterion -> regex -> regex-automata`

```
[dev-dependencies]
├── criterion v0.4.0
│   ├── anes v0.1.6
│   ├── atty v0.2.14
│   │   └── libc v0.2.148
│   ├── cast v0.3.0
│   ├── ciborium v0.2.1
│   │   ├── ciborium-io v0.2.1
│   │   ├── ciborium-ll v0.2.1
│   │   │   ├── ciborium-io v0.2.1
│   │   │   └── half v1.8.2
│   │   └── serde v1.0.188 (*)
│   ├── clap v3.2.25
│   │   ├── bitflags v1.3.2
│   │   ├── clap_lex v0.2.4
│   │   │   └── os_str_bytes v6.5.1
│   │   ├── indexmap v1.9.3 (*)
│   │   └── textwrap v0.16.0
│   ├── criterion-plot v0.5.0
│   │   ├── cast v0.3.0
│   │   └── itertools v0.10.5
│   │       └── either v1.9.0
│   ├── itertools v0.10.5 (*)
│   ├── lazy_static v1.4.0
│   ├── num-traits v0.2.16 (*)
│   ├── oorandom v11.1.3
│   ├── plotters v0.3.5
│   │   ├── num-traits v0.2.16 (*)
│   │   ├── plotters-backend v0.3.5
│   │   └── plotters-svg v0.3.5
│   │       └── plotters-backend v0.3.5
│   ├── rayon v1.7.0 (*)
│   ├── regex v1.9.5
│   │   ├── regex-automata v0.3.8
│   │   │   └── regex-syntax v0.7.5
│   │   └── regex-syntax v0.7.5
│   ├── serde v1.0.188 (*)
│   ├── serde_derive v1.0.188 (proc-macro) (*)
│   ├── serde_json v1.0.107 (*)
│   ├── tinytemplate v1.2.1
│   │   ├── serde v1.0.188 (*)
│   │   └── serde_json v1.0.107 (*)
│   └── walkdir v2.4.0
│       └── same-file v1.0.6
```

File: .readthedocs.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ build:
     - cmake
   tools:
     python: "3.11"
-    rust: "1.64"
+    rust: "1.70"  # RDT does not offer our minimal version 1.65
```
Comment (Luthaf) [inline .readthedocs.yml:15]: Should we just use the latest rustc version on rtd?

**Comment by Luthaf** [comment]
Ok, it seems painful to remove regex from criterion, so we can bump!

File: .readthedocs.yml (hunk 1 above)
Comment (Luthaf) [reply .readthedocs.yml:15]: ok, this is the latest version. I would just remove the comment about the minimal version then.

Comment (Luthaf) [reply .readthedocs.yml:15]:
```suggestion
    rust: "1.70"
```

**Comment by PicoCentauri** [comment]
Bump it is!

**Review by Luthaf — APPROVED** [review]

**Pull request merged**